src/wallet/wallet.cpp

```diff
@@ -2442,7 +2442,11 @@ void CWallet::AvailableCoins(interfaces::Chain::Lock& locked_chain, std::vector<
             continue;
         }
 
-        if (nDepth < nMinDepth || nDepth > nMaxDepth)
+        int min_depth = 0;
+        if (coinControl)
+            min_depth = coinControl->m_min_depth;
```

Comment:
Instead of running this once per loop, could move it to the beginning of the function (out of the loop). That would also avoid hiding the default value `0` deep in the function body.

```cpp
const int min_depth{coinControl?coinControl->m_min_depth:0};